fitbenchmarking/results_processing/performance_profiler.py

```diff
@@ -382,17 +442,105 @@ def layout(self):
         :rtype: dcc.Graph
         """
 
-        return dcc.Graph(id=f"visual {self.identif}")
+        return dcc.Graph(id=f"visual {self.id}")
 
     def set_callbacks(self):
         """Calls callbacks on the function that creates the dash graph."""
 
         dash.callback(
-            Output(f"visual {self.identif}", "figure"),
+            Output(f"visual {self.id}", "figure"),
             [Input("Log axis toggle", "value"),
              Input("dropdown", "value")]
         )(self.create_graph)
 
+    def update_linestyles(self, solvers) -> None:
+        """
+        Determines a combination of linestyle and color for each solver.
+
+        :param solvers: Solvers to be selected, max 15
+        :type solvers: list[str]
+        """
+        previous_solvers = list(self.current_styles.keys())
+
+        # If this is the first time executing the code
+        if len(previous_solvers) == 0:
+            for i, solver in enumerate(solvers):
+                linestyle = self.linestyles[(i % len(self.linestyles))]
+                linecolor = self.colors[(i % len(self.colors))]
+                self.current_styles[solver] = {
+                    'linestyle': linestyle,
+                    'linecolor': linecolor,
+                }
+                comb_id = linestyle+linecolor
+                del self.avail_styles[comb_id]
+
+        # If a solver has been added
+        elif len(solvers) > len(previous_solvers):
+
+            newly_added_solvers = set(solvers).difference(previous_solvers)
+
+            for solver in newly_added_solvers:
+                chosen_style = list(self.avail_styles.values())[-1]
+                self.current_styles[solver] = {
+                    'linestyle': chosen_style[0],
+                    'linecolor': chosen_style[1],
+                }
+                comb_id = chosen_style[0]+chosen_style[1]
+                del self.avail_styles[comb_id]
+
+        # If a solver has been removed
+        elif len(solvers) < len(previous_solvers):
+
+            solvers_to_remove = set(previous_solvers).difference(solvers)
+
+            for solver in solvers_to_remove:
+                linestyle = self.current_styles[solver]['linestyle']
+                linecolor = self.current_styles[solver]['linecolor']
+                self.avail_styles[linestyle+linecolor] = (linestyle, linecolor)
+                self.current_styles.pop(solver)
```

Comment:
`pop` gives the values and removes it at the same time. 
```suggestion
                comb = self.current_styles.pop(solver)
                self.avail_styles.append(comb)
                
```